routers/stops.py

```diff
@@ -1,66 +1,86 @@
 from fastapi import status, APIRouter, HTTPException, Depends
 
 from dependencies import get_current_user
-from models import supabase, Stop, User, UserEntity
+from models import supabase, Stop, User, UserEntity, StopEntity
 
 router = APIRouter(prefix="/stops", tags=["stops"])
 
-#TODO please check the logic here
-#TODO while creating a stop as an operator, I get a bad request error but stop is still created
 # Define the endpoint for creating a stop
 @router.post("/stop")
-def create_stop(long: str, lat: str, current_user: User = Depends(get_current_user)):
-    # Check if the current user is a passenger or a parcel operator
-    if current_user.entity == UserEntity.passenger or current_user.entity == UserEntity.parcel_operator:
-        if current_user.entity == UserEntity.passenger:
-            # Check if the passenger already requested a stop
-            # Query the stop table with user id
-            response = supabase.table("stops").select("*").eq("user_id", current_user.id).execute()
-            # Check if response has data
-            if response.data:
-                # Raise an exception if the stop is not possible
+def create_stop(stop : Stop, current_user: User = Depends(get_current_user)):
+    # Check if the current user is a driver
+    if current_user.entity == UserEntity.driver:
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail="Drivers can't create stops",
+        )
+    else:
+        if stop.entity == StopEntity.static:
+            if current_user.entity != UserEntity.admin:
```

Comment:
Who admin

Reply:
I understand the concern that we don't really have a user type to create the bus stops, but I'd rather tie this functionality to parcel manager

Reply:
In any case the name admin is probably too ambiguous